### DISCUSSION TITLE
Exercise 11: Show that, if channels are not FIFO, then Chandy–Lamport snapshot algorithm does not work.

### DISCUSSION THREAD

**Discussion by dag7dev** [sapienzastudentsnetwork/distributed-systems#36] [Exercises]
**Exercise 11:** Show that, if channels are not FIFO, then Chandy–Lamport snapshot algorithm does not work.

**Comment by susannacifani** [comment]
Let's assume that channels aren't FIFO, so we could have that the messages from the same sender aren't delivered in the same order in which they were sent, so: $e \rightarrow e' \land e' \in S \nRightarrow e \in S$.
That means that we could have an inconsistent global state, and that the Chandy-Lamport protocol could not work.

On Drive they also make an example: For example, it can happen that $p1$ sends its “take snapshot” $e$ and after that it sends a message $e'$. Then $p2$ receives first $e'$ and then $e$. So, $e' \in C$, $e \rightarrow e'$, but $e \notin C$.
